packages/utils/src/provider_utils.ts

```diff
@@ -42,6 +42,10 @@ export const providerUtils = {
             enable: (supportedProvider as any).enable,
             sendAsync: _.noop.bind(_), // Will be replaced
         };
+        if (provider.enable) {
+            // Need to bind, metamask can lose reference to function without binding as of 7.7.0
+            provider.enable.bind(supportedProvider)
```

Comment:
I think bind creates a new fn so you may need to reassign `provider.enable`
```suggestion
            provider.enable = provider.enable.bind(supportedProvider)
```

Reply:
blah! that's what I meant to do :) thanks